### PR TITLE
Feature/import addresses for province of southwest finland

### PR DESCRIPTION
**Pull request by juuso-j**
# Support for multiple provinces and added province Southwest Finland

#### Scripts
1. scripts/import-digiroad-data.sh
    * Now imports addresses by the the province given as argument.  
2. scripts/import-paavo-data.sh
    * Now imports postal codes and offices by the province given as argument.
#### Importers
1. address/services/address_import.py
    * Moved import functionality into a class. The constructor of the class takes the province as input, this variable is used when deleting or importing by province. Added province names as keys to the MUNICIPALITIES dict. Also added the municipalities of Southwest Finland. 
4. address/services/postal_code_import.py
    * Moved functionality into a class to make importer uniform with the address importer.
    
#### Commands
1. address/management/commands/import_addresses.py
    * Now uses the instance with the province argument of the AddressImporter class to delete and import addresses.
5. address/management/commands/import_postal_codes.py
    * Now uses the instance of the PostalCode importer class to import. 

#### Tests
1. address/tests/test_address_import_service.py
    * Changed to use instances of AddressImporter class with "uusima" as province.  
2. address/tests/test_import_addresses_command.py
    * Added the province to the call_command.
4. address/tests/test_postal_code_import_service.py
    * Changed to use instances of PostalCodeImporter.

### CODE REVIEW

File: scripts/import-digiroad-data.sh [hunk 1]
```diff
@@ -1,12 +1,32 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e
+available_provinces="'uusimaa' and 'varisinais-suomi'"
```
Comment (mhieta) [inline scripts/import-digiroad-data.sh:4]: Fix typo `varisinais-suomi`

Comment (juuso-j) [reply scripts/import-digiroad-data.sh]: fixed.

File: scripts/import-digiroad-data.sh [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
+    echo "Province $1 not found, available provinces are ${available_provinces}."; 
+    exit 0
+fi
+
+echo ${package_name[$1]}
+
+DATA_URL="https://ava.vaylapilvi.fi/ava/Tiest%C3%B6tiedot/Digiroad/Digiroad-irrotusaineistot/latest/Maakuntajako_DIGIROAD_R_EUREF-FIN/${package_name[$1]}"
 # Directory where the data will be downloaded and extracted
-DATA_DIR=/tmp/digiroad
+DATA_DIR=/home/juuso/repos/geo-search/scripts/tmp/digiroad
```
Comment (mhieta) [inline scripts/import-digiroad-data.sh]: I think this should be reverted back to use the regular `/tmp/digiroad`-folder

File: scripts/import-digiroad-data.sh [hunk 3]
```diff
@@ -23,12 +43,12 @@ CONVERTED_DIR=$DATA_DIR/converted
 mkdir -p $DATA_DIR
 curl "$DATA_URL" -o $DATA_DIR/data.zip
 
-# Extract the shapefiles from the archive
+# # Extract the shapefiles from the archive
```
Comment (mhieta) [inline scripts/import-digiroad-data.sh:46]: Remove this duplicate `#`

File: scripts/import-paavo-data.sh [hunk 4]
```diff
@@ -1,12 +1,32 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e
 
+available_provinces="'uusimaa' and 'varisinais-suomi'"
```
Comment (mhieta) [inline scripts/import-paavo-data.sh]: Fix typo here

File: scripts/import-digiroad-data.sh (hunk 2 above)
Comment (juuso-j) [reply scripts/import-digiroad-data.sh]: That has been done.

File: scripts/import-paavo-data.sh [hunk 5]
```diff
@@ -1,12 +1,32 @@
...
+    exit 0
+fi
+
 # URL to Paavo WFS service
-DATA_URL="http://geo.stat.fi/geoserver/wfs?SERVICE=wfs&version=1.0.0&request=GetFeature&srsName=EPSG:3067&outputFormat=SHAPE-ZIP&typeNames=pno_meri_2021&bbox=267392.57814054575,6636189.158504381,476288.57814054575,6747293.158504381"
+DATA_URL="http://geo.stat.fi/geoserver/wfs?SERVICE=wfs&version=1.0.0&request=GetFeature&srsName=EPSG:3067&outputFormat=SHAPE-ZIP&typeNames=pno_meri_2021&bbox=${bbox[$1]}"
 
-# Directory where the data will be downloaded and extracted
-DATA_DIR=/tmp/paavo
+DATA_DIR=/home/juuso/repos/geo-search/scripts/tmp/paavo
```
Comment (mhieta) [inline scripts/import-paavo-data.sh]: Revert back to `/tmp/paavo`

File: scripts/import-digiroad-data.sh [hunk 6]
```diff
@@ -7,7 +7,7 @@ if [ $# == 0 ]
 then
     echo "No province argument supplied."    
     echo "Available provinces are ${available_provinces}."
-    exit 0
+    exit p0
```
Comment (mhieta) [inline scripts/import-digiroad-data.sh:10]: Should this be still `exit 0` ?

Comment (juuso-j) [reply scripts/import-digiroad-data.sh:10]: yes, it should be, will fix.

**Review by mhieta — COMMENTED** [review]
Added couple of comments. If you want to aim per perfectness, then also remember to not use any trailing comma in the end of commit message header and also keep the commit body lines max 72 chars long (source https://cbea.ms/git-commit/)

**Review by mhieta — APPROVED** [review]
Excellent work! 💯 
Approved this PR now.

**Pull request merged**